src/Domain/LogicalGroup.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Supervisorg\Domain;
+
+class LogicalGroup
+{
+    private
+        $name,
+        $icon,
+        $default,
+        $regex;
+
+    public function __construct($name, array $definition)
+    {
+        $this->name = $name;
+
+        $definition = $this->ensureDefinitionIsValid($definition);
+
+        $this->regex = sprintf('~%s~', $definition['regex']);
+        $this->icon = $definition['icon'];
+        $this->default = $definition['defaultView'] === true;
+    }
+
+    private function ensureDefinitionIsValid(array $definition)
+    {
+        $defaultValues = [
+            'defaultView' => false,
+            'icon' => 'circle-o-notch',
+        ];
+
+        $definition = $definition + $defaultValues;
+
+        $requiredFields = ['regex'];
+        foreach($requiredFields as $field)
+        {
+            if(! isset($definition[$field]))
+            {
+                throw new \InvalidArgumentException(sprintf(
+                    'Missing field %s in logical group %s definition',
+                    $field,
+                    $this->name
+                ));
+            }
+        }
+
+        return $definition;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function getIcon()
+    {
+        return $this->icon;
+    }
+
+    public function isDefault()
+    {
+        return $this->default;
+    }
+
+    public function belongTo(Process $process, $value)
```

Comment:
I thought about it but we quite never use the 3rd person for verbs in method name (ensureX, buildX, computeX, ...)